### PR TITLE
Provide error messages for Otp-enroll component received from backend

### LINKED ISSUE #94: Provide better error messages for Otp-enroll component
Here we have our OTP-Enroll component https://github.com/coding-blocks/codingblocks.online.projectx/blob/master/app/pods/components/otp-enroll/component.js

It is concerned with two `ember-concurrency` tasks: `sendOtpTask` and `verifyOtpTask`. We would want to provide error messages to the user as received from the backend instead hard-coded one. The error ~~can be a string or a~~ is a serialized Error object. The backend response code in case of error will be 400/500.

For VerifyOtp if you get 500, simple show "Wrong Otp". For 400 you will get an Error object, use that message.

### PR DESCRIPTION
**Pull request by architkshk**
fixes #94 @abhishek97

### CODE REVIEW

File: app/pods/components/otp-enroll/component.js [hunk 1]
```diff
@@ -27,7 +27,10 @@ export default class otpEnrollComponent extends Component {
     })
     .catch(err => {
       console.error(err)
-      this.set('errorString', 'Cannot Send OTP to that email. Please use your registered email.')
+      const errorMsg = JSON.parse(err);
```
Comment (abhishek97) [inline app/pods/components/otp-enroll/component.js]: If `err` is not a JSON string, then `JSON.parse` here will throw an error. Try using

```js
try { 
   ...
} catch (e) { 
   ... 
}

```

File: app/pods/components/otp-enroll/component.js [hunk 2]
```diff
@@ -42,7 +45,10 @@ export default class otpEnrollComponent extends Component {
       window.location.reload()
     }).catch(err => {
       console.error(err)
-      this.set('errorString', 'Incorrect OTP');
+      const errorMsg = JSON.parse(err);
+      if (typeof errorMsg == Object)
```
Comment (abhishek97) [inline app/pods/components/otp-enroll/component.js]: keep your code DRY, extract this to a util/method maybe?

Comment (architkshk) [reply app/pods/components/otp-enroll/component.js]: done

**Comment by architkshk** [comment]
@abhishek97 please check

**Comment by abhishek97** [comment]
@architkshk Looks good, However let me keep this on hold for now. We will be changing backend to return `Error` instances only, so that you can just do `err.message`

**Comment by architkshk** [comment]
Ok

**Comment by abhishek97** [comment]
@architkshk I have made appropriate changes on the backend so that you will get a serialized Error object in case of any error. Please Re-do this PR with appropriate changes and we'll merge that. Thanks.

**Pull request closed without merging**